code/Test_definitions/LocationRetrieval.feature

```diff
@@ -0,0 +1,83 @@
+Feature: Get a device location (POST /retrieve)
+
+  @location_retrieval_01_by_phoneNumber_or_ipv4Address_or_networkAccessIdentifier
+  Scenario: Get the location of a device
+    Given a valid device <device> identified by a <deviceIdentierName> <deviceIdentierValue>
+    And a provided maxAge of 80 seconds
+    When a retrieve device location request is made 
+    Then the response status code should be 200
+    And the response body should provide a valid area 
+
+    Examples:
+        | deviceIdentifierName     |  deviceIdentifierValue                 |   
+        | phoneNumber              |  +34666111333                          |   
+        | networkAccessIdentifier  |  123456789@domain.com                  |   
+        | ipv6Address              |  2001:db8:85a3:8d3:1319:8a2e:370:7344  |   
+

```

Comment:
Should we have testcases for when a location cannot be returned ? This can happen in two ways:

- The network does not have a location record for the device at all, e.g. it was trurned off a while ago.
- There is a location information in the network, but it does not satisfy the maxAge criterion.

Reply:
Agreed - added Test Case

Reply:
We can add the fact that the location is known by the operator to the Given steps